metagpt/roles/role.py

```diff
@@ -238,3 +270,33 @@ async def run(self, message=None):
         # 将回复发布到环境，等待下一个订阅者处理
         self._publish_message(rsp)
         return rsp
+
+    @staticmethod
+    def supply_options(options):
+        """Supply missing options"""
+        ret = Role.__DEFAULT_OPTIONS__.copy()
+        if not options:
+            return ret
+        ret.update(options)
+        return ret
+
+    @staticmethod
+    def format_value(value, options):
+        """Fill parameters inside `value` with `options`.
+        """
+        if "{" not in value:
```

Comment:
if `value` contains `{}`, there might be variable placeholders inside; otherwise it's plain text. 
For example:
```
You are a {teaching_language} teacher.
```